### PR TITLE
Upgrading library for production

### PR DESCRIPTION
**Pull request by madhephaestus**
In this PR i am adding all of the changes to use this servo with all of its features. 

The first update is to add electrical connection documentation. 

The printf() functon is replaced with the Arduino compatible Serial.printf().

Formatting to adhere to C++ conventions. 

Implement as many commands as possible.

### CODE REVIEW

File: src/lx16a-servo.cpp [hunk 1]
```diff
@@ -4,96 +4,145 @@
...
-    buf[1] = 0x55;
-    buf[2] = _id;
-    buf[3] = buflen-3;
-    buf[4] = cmd;
-    for (int i=0; i<param_cnt; i++) buf[5+i] = params[i];
-    uint8_t cksum = 0;
-    for (int i=2; i<buflen-1; i++) cksum += buf[i];
-    buf[buflen-1] = ~cksum;
+	if (param_cnt < 0 || param_cnt > 4)
+		return false;
```
Comment (tve) [inline src/lx16a-servo.cpp:7]: Quick comment here: I'm OK with
```
if (condition) statement;
```
or
```
if (condition) {
    statement;
}
```
but not the multiline version without braces 'cause it's too easy in a future edit to make a mistake.

**Comment by tve** [comment]
Hey, if you could like to take over this library I'm happy to put a link to yours into the readme in my repo. I still have HW running this but I will be converting it to MicroPython in the coming months. As such, I won't have time to test PRs and the library, as I have it, works for my needs for now.

**Comment by madhephaestus** [comment]
I wouldn't mind making my fork the main branch included in Arduino Library manager, i just wouldn't want to take the credit away from you :)

Thank you so much for the hard work getting it started. I am going through and applying Arduino and C++ conventions at the moment, but intend to switch over to adding commands and making this a full-feature library for use with these motors.

**Comment by tve** [comment]
I'd be happy to point to yours. Just leave a pointer back to mine as original/initial/... version somewhere.

**Comment by madhephaestus** [comment]
I will keep the source history and the GitHub link back to the original. Ill start adding it to Arduino now :D

**Pull request closed without merging**